src/esp/physics/bullet/BulletArticulatedObject.cpp

```diff
@@ -722,5 +697,24 @@ void BulletArticulatedObject::clampJointLimits() {
   }
 }
 
+void BulletArticulatedObject::updateKinematicState() {
+  btAlignedObjectArray<btQuaternion> scratch_q;
+  btAlignedObjectArray<btVector3> scratch_m;
+  btMultiBody_->forwardKinematics(scratch_q, scratch_m);
```

Comment:
Nit: would be nice to see these scratch buffers persist rather than get allocated/deallocated for every call.